include/gz/math/Cone.hh

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2024 CogniPilot Foundation
+ * Copyright 2024 Open Source Robotics Foundation
+ * Copyright 2024 Rudis Laboratories
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_MATH_CONE_HH_
+#define GZ_MATH_CONE_HH_
+
+#include <optional>
+#include "gz/math/MassMatrix3.hh"
+#include "gz/math/Material.hh"
+#include "gz/math/Quaternion.hh"
+
+namespace gz
+{
+  namespace math
+  {
+    // Foward declarations
+    class ConePrivate;
+
+    // Inline bracket to help doxygen filtering.
+    inline namespace GZ_MATH_VERSION_NAMESPACE {
+    //
+    /// \class Cone Cone.hh gz/math/Cone.hh
+    /// \brief A representation of a cone.
+    ///
+    /// The cone class supports defining a cone with a radius,
+    /// length, rotational offset, and material properties. Radius and
+    /// length are in meters. See Material for more on material properties.
+    /// By default, a cone's length is aligned with the Z axis where the base
+    /// of the cone is proximal to the origin and vertex points in positive Z.
+    /// The rotational offset encodes a rotation from the z axis.
+    template<typename Precision>
+    class Cone
+    {
+      /// \brief Default constructor. The default radius and length are both
+      /// zero. The default rotational offset is
+      /// Quaternion<Precision>::Identity.
+      public: Cone() = default;
+
+      /// \brief Construct a cone with a length, radius, and optionally
+      /// a rotational offset.
+      /// \param[in] _length Length of the cone.
+      /// \param[in] _radius Radius of the cone.
+      /// \param[in] _rotOffset Rotational offset of the cone.
+      public: Cone(const Precision _length, const Precision _radius,
```

Comment:
I think we should remove all `const` from `Precision`. This is a primitive, and we don't use `const` with basic primitives. @scpeters or @azeey, thoughts ?

Reply:
@ahcorde I think that is just a copy/paste from Cylinder with minor changes FYI:
https://github.com/gazebosim/gz-math/blob/2f1e049bc6937efab3fd7a1d136978fc3339cf30/include/gz/math/Cylinder.hh#L51-L157

Reply:
And if we change it in Cone we would probably need to change it in a lot of other places?

Reply:
I also just pushed a PR to fix the linting issues I think popped up from the last "batch suggestion apply". Should all be green after that.

Reply:
address the `const` issues in a separate PR where we can make multiple changes in a consistent way

Reply:
@scpeters any other changes desired/needed on this particular PR?